### PR TITLE
CTC-53 & CTC-54: OakMediaClip, TagFunctional and Timer

### PR DESCRIPTION
**Pull request by k-huggs**
# How to review this PR

_Leave this text block for the reviewer_

- Check [component hierarchy](https://miro.com/app/board/uXjVNnKBgyk=/?share_link_id=59445593794) is followed correctly
- Check the design [Heuristics](https://components.thenational.academy/?path=/docs/docs-howtodesigncomponents--docs#heuristics-for-component-design) have been followed
- Check [naming conventions](https://components.thenational.academy/?path=/docs/docs-namingconventions--docs) have been applied
- Check for these gotchyas:
  - Missing exports for Oak components
  - Accidental export of Internal components
  - Snapshots of unexpected components have been modified
  - Circular dependencies
  - Code duplication (via not using base components)
  - Non-functional storybook for this or related components
  - Sensitve files changed eg. atoms, or style tokens
  - Relative imports
  - Default exports

# Add your PR description below

### CTC-53
Create `OakMediaClip` component 

### CTC-54
Create `OakMediaClip` subcomponents: `OakTimer` & `OakTagFunctional`

## Link to the design doc

https://www.figma.com/design/iTimVTBZ3XcszhRa9KtD7r/Cycle-2-requirements---Teachers?node-id=124-20501&node-type=frame&m=dev

#### Oak Media Clip
https://deploy-preview-324--lively-meringue-8ebd43.netlify.app/?path=/docs/components-organisms-teacher-oakmediaclip--docs

#### Oak Tag Functional
https://deploy-preview-324--lively-meringue-8ebd43.netlify.app/?path=/docs/components-molecules-oaktagfunctional--docs

#### Oak Timer
https://deploy-preview-324--lively-meringue-8ebd43.netlify.app/?path=/docs/components-molecules-oaktimer--docs

### CTC-53
OakMediaClip should have different states viewable, clickable, enter learningCycle to render under the Clip name, to see both with and without the learning cycle text

### CTC-53

- [ ]  pass through thumbnail image (path , or some mux reference)
- [ ]  time code
- [ ]  Clip name
- [ ]  learning cycle (toggleable)
- [ ]  standard / playing / played
- [ ]  onClick
- [ ]  disabled state
- [ ]  image alt text ?**

### CTC-54

- [ ]  timer block - mins and seconds
- [ ]  played block

### CODE REVIEW

**Comment by weronika-szalas** [comment]
Looks good from coding perspective, will re-review after the QA comments updates.

File: src/components/molecules/OakTimer/OakTimer.tsx [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  const formattedTimeCode = formatTimeCode(timeCode);
+
+  return (
+    <StyledTimeBox
+      $color={"text-inverted"}
+      $borderRadius={"border-radius-xs"}
+      $background={"bg-icon"}
+      $font={["body-4"]}
+      {...oakBoxProps}
+    >
```
Comment (RBrownsett) [inline src/components/molecules/OakTimer/OakTimer.tsx:33]: @k-huggs would it be possible to add something like the below to improve the accessibility for screen readers? currently it just reads the number without any context to what that number is. 
`aria-label={`Media clip length: ${formattedTimeCode}`}`

**Comment by RBrownsett** [comment]
> @RBrownsett what is the smallest mobile size we test on? We design for 375px and it's ok on this size. Otherwise, I'd say at certain view port size (e.g. 360px) wrap the played label under the titles. One change from me:
> 
> * [ ]  can learning cycles be truncated to one line? This way long titles won't take up too much space on mobile.

Good point - this is Storybooks inbuilt 'small mobile' size which is a lot smaller at 320px. Like you said it looks ok on 375px so I imagine we can ignore the issue and double check when the page is put together using browserstack

**Review by RBrownsett — APPROVED** [review]
Approved - changes made, components tested with no issues.

File: src/components/molecules/OakCollapsibleContent/__snapshots__/OakCollapsibleContent.test.tsx.snap [hunk 2]
```diff
@@ -22,6 +22,7 @@ exports[`OakCollapsibleContent matches snapshot 1`] = `
   max-height: 100%;
   padding-right: 1rem;
   font-family: --var(google-font),Lexend,sans-serif;
+  overflow: auto;
```
Comment (k-huggs) [inline src/components/molecules/OakCollapsibleContent/__snapshots__/OakCollapsibleContent.test.tsx.snap:25]: Some of these snapshots automatically updated when I added overflow to the typography styles

**Review by weronika-szalas — APPROVED** [review]
Looks good!

**Pull request merged**